luigi/contrib/gcs.py

```diff
@@ -370,7 +370,12 @@ def download(self, path, chunksize=None):
 class _DeleteOnCloseFile(io.FileIO):
     def close(self):
         super(_DeleteOnCloseFile, self).close()
-        os.remove(self.name)
+        if not self.closed:
```

Comment:
@Tarrasch Good point.  It's a standard property on `io.FileIO` that is always set to `True` or `False` depending on the state of the object.  I switched to just a `try / except`.